### PR TITLE
Q3 small signin updates

### PR DESCRIPTION
**Pull request by besiler**
Schema PR: 5649693

Added
* Clientcredentialtype property to signins
* RemoteDesktopToken to incomingtokentype
* Confirmcompromised API
* Confirmsafe API

### CODE REVIEW

File: changelog/Microsoft.AAD.Reporting.json [hunk 1]
```diff
@@ -1,5 +1,55 @@
...
+          "Description": "Added the [confirmCompromised](https://docs.microsoft.com/en-us/graph/api/confirmCompromised?view=graph-rest-beta) method to the [confirmCompromised](https://docs.microsoft.com/en-us/graph/api/confirmCompromised?view=graph-rest-beta) resource.",
+          "Target": "confirmCompromised"
+        },
+        {
+          "Id": "6a60b3dd-cb30-4a9b-8b94-9197e17d4ba5",
+          "ApiChange": "Method",
+          "ChangedApiName": "confirmSafe",
+          "ChangeType": "Addition",
+          "Description": "Added the [confirmSafe](https://docs.microsoft.com/en-us/graph/api/confirmSafe?view=graph-rest-beta) method to the [confirmSafe](https://docs.microsoft.com/en-us/graph/api/confirmSafe?view=graph-rest-beta) resource.",
+          "Target": "confirmSafe"
```
Comment (FaithOmbongi) [inline changelog/Microsoft.AAD.Reporting.json:43]:
```suggestion
          "Description": "Added the [confirmCompromised](https://docs.microsoft.com/en-us/graph/api/confirmCompromised?view=graph-rest-beta) method to the [signin](https://docs.microsoft.com/en-us/graph/api/signin?view=graph-rest-beta) resource.",
          "Target": "signIn"
        },
        {
          "Id": "6a60b3dd-cb30-4a9b-8b94-9197e17d4ba5",
          "ApiChange": "Method",
          "ChangedApiName": "confirmSafe",
          "ChangeType": "Addition",
          "Description": "Added the [confirmSafe](https://docs.microsoft.com/en-us/graph/api/confirmSafe?view=graph-rest-beta) method to the [signIn](https://docs.microsoft.com/en-us/graph/api/signin?view=graph-rest-beta) resource.",
          "Target": "signIn"
```

File: changelog/Microsoft.AAD.Reporting.json [hunk 2]
```diff
@@ -1,5 +1,55 @@
...
+        }
+      ],
+      "Id": "6a60b3dd-cb30-4a9b-8b94-9197e17d4ba5",
+      "Cloud": "Prod",
+      "Version": "beta",
+      "CreatedDateTime": "2022-03-25T21:43:53.6754321Z",
+      "WorkloadArea": "Reports and audit",
```
Comment (FaithOmbongi) [inline changelog/Microsoft.AAD.Reporting.json:50]:
```suggestion
      "WorkloadArea": "Reports",
```

File: api-reference/beta/api/signin-confirmcompromised.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
+---
+title: "signIn: confirmCompromised"
+description: "Allows you to mark Azure AD log events as risky for Azure AD Identity Protection."
```
Comment (FaithOmbongi) [inline api-reference/beta/api/signin-confirmcompromised.md:3]: I think not explicitly mentioning sign ins infers this action is also for other log types.
```suggestion
description: "Allows you to mark Azure AD sign in events as risky for Azure AD Identity Protection."
```

File: api-reference/beta/api/signin-confirmcompromised.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+ms.prod: "identity-and-access-reports"
+doc_type: apiPageType
+---
+
+# signIn: confirmCompromised
+Namespace: microsoft.graph
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+This API allows admins to mark an event in the Azure AD logs as risky. Events marked as risky by an admin are immediately flagged as high risk in Azure AD Identity Protection, overriding previous risk states. Admins can confirm that events flagged as risky by Azure AD Identity Protection are in fact risky or can mark unflagged events as risky. To mark a sign-in as risky, include he requestID of the sign-in in the request body. If successful, this method returns a 204 No Content response code. It does not return anything in the response body.
```
Comment (FaithOmbongi) [inline api-reference/beta/api/signin-confirmcompromised.md:15]: Suggestion to remove the last 2 sentences. Those will be catered for in the Request body and Response sections below.
```suggestion
Allows admins to mark an event in the Azure AD sign in logs as risky. Events marked as risky by an admin are immediately flagged as high risk in Azure AD Identity Protection, overriding previous risk states. Admins can confirm that events flagged as risky by Azure AD Identity Protection are in fact risky or they can mark unflagged events as risky. 
```

File: api-reference/beta/api/signin-confirmcompromised.md [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+### Request
+<!-- {
+  "blockType": "request",
+  "name": "signinthis.confirmcompromised"
+}
+-->
+``` http
+POST https://graph.microsoft.com/beta/auditLogs/signIns/confirmCompromised
+Content-Type: application/json
+Content-length: 42
```
Comment (FaithOmbongi) [inline api-reference/beta/api/signin-confirmcompromised.md:68]: Let's remove this.
```suggestion
```

File: api-reference/beta/api/signin-confirmcompromised.md [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+    "f01c6af6-6683-4a37-a945-0a925501eede",
+    "42bf60ac-d0cb-4206-aa5c-101884298f55",
+    "f09c8f14-8d8e-42cf-8a7e-732b0594e79b"
+  ]
+}
+```
+
+
+### Response
+>**Note:** The response object shown here might be shortened for readability.
```
Comment (FaithOmbongi) [inline api-reference/beta/api/signin-confirmcompromised.md:81]:
```suggestion
```

File: api-reference/beta/api/signin-confirmsafe.md [hunk 7]
```diff
@@ -0,0 +1,90 @@
+---
+title: "signIn: confirmSafe"
+description: "Allows you to mark Azure AD log events as safe for Azure AD Identity Protection"
```
Comment (FaithOmbongi) [inline api-reference/beta/api/signin-confirmsafe.md:3]:
```suggestion
description: "Allows you to mark Azure AD sign in events as safe for Azure AD Identity Protection."
```

File: api-reference/beta/api/signin-confirmsafe.md [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+ms.prod: "identity-and-access-reports"
+doc_type: apiPageType
+---
+
+# signIn: confirmSafe
+Namespace: microsoft.graph
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+This API allows admins to mark an event in the Azure AD logs as safe. Admins can either mark that events flagged as risky by Azure AD Identity Protection are in fact safe or can mark unflagged events as safe if needed. To mark a sign-in as safe, include he requestID of the sign-in in the request body. If successful, this method returns a 204 No Content response code. It does not return anything in the response body.
```
Comment (FaithOmbongi) [inline api-reference/beta/api/signin-confirmsafe.md:15]:
```suggestion
Allows admins to mark an event in Azure AD sign in logs as safe. Admins can either mark as safe the events flagged as risky by Azure AD Identity Protection or they can mark unflagged events as safe.
```

File: api-reference/beta/api/signin-confirmsafe.md [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+|Content-Type|application/json. Required.|
+
+## Request body
+In the request body, supply JSON representation of the parameters.
+
+The following table shows the parameters that can be used with this action.
+
+|Parameter|Type|Description|
+|:---|:---|:---|
+|requestIds|String collection|The requestIds of the events that should be marked safe for Azure AD Identity Protection.|
```
Comment (FaithOmbongi) [inline api-reference/beta/api/signin-confirmsafe.md:49]: I think you're referring to the values in the **id** property?
```suggestion
|requestIds|String collection|The IDs of the sign in events that should be marked safe for Azure AD Identity Protection.|
```

File: api-reference/beta/api/signin-confirmcompromised.md [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+|Content-Type|application/json. Required.|
+
+## Request body
+In the request body, supply JSON representation of the parameters.
+
+The following table shows the parameters that can be used with this action.
+
+|Parameter|Type|Description|
+|:---|:---|:---|
+|requestIds|String collection|The requestIds of the events that should be marked risky for Azure AD Identity Protection.|
```
Comment (FaithOmbongi) [inline api-reference/beta/api/signin-confirmcompromised.md:49]: I believe you're referring to the value in the **id** property?
```suggestion
|requestIds|String collection|The IDs of the sign in events that should be marked risky for Azure AD Identity Protection.|
```

File: api-reference/beta/api/signin-confirmsafe.md [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+### Request
+<!-- {
+  "blockType": "request",
+  "name": "signinthis.confirmsafe"
+}
+-->
+``` http
+POST https://graph.microsoft.com/beta/auditLogs/signIns/confirmSafe
+Content-Type: application/json
+Content-length: 42
```
Comment (FaithOmbongi) [inline api-reference/beta/api/signin-confirmsafe.md:68]:
```suggestion
```

**Review by FaithOmbongi — CHANGES_REQUESTED** [review]
Thank you @besiler for your work on this PR. I've left suggestions for some minor fixes and please surface the new APIs in the TOC.

**Review by FaithOmbongi — CHANGES_REQUESTED** [review]
Thank you @besiler for your work on this PR. I've left suggestions for some minor fixes and please surface the new APIs in the TOC.

File: api-reference/beta/api/signin-confirmcompromised.md (hunk 10 above)
Comment (besiler) [reply api-reference/beta/api/signin-confirmcompromised.md:49]: Yep, that's correct

**Comment by besiler** [comment]
> Thank you @besiler for your work on this PR. I've left suggestions for some minor fixes and please surface the new APIs in the TOC.

How do I add the new APIs to the TOC? If you have a wiki doc or something I can look at to add them, that would help!

I looked at the TOC file under beta > breadcrumb > toc but don't see much on how I would add the new APIs.

image

**Comment by FaithOmbongi** [comment]
> 

Hi @besiler

**Review by FaithOmbongi — APPROVED** [review]
Thank you @besiler for your work on this PR. Approved.

**Pull request merged**